pkg/render/common.go

```diff
@@ -371,3 +371,23 @@ func basePodSecurityPolicy() *policyv1beta1.PodSecurityPolicy {
 		},
 	}
 }
+
+// K8sServiceEndpoint is the Host/Port of the K8s endpoint.
+type K8sServiceEndpoint struct {
+	Host string
+	Port string
+}
+
+// EnvVars returns a slice of v1.EnvVars KUBERNETES_SERVICE_HOST/PORT if the Host and Port
+// of the K8sServiceEndpoint were set. It returns a nil slice if either was empty as both
+// need to be set.
+func (k8s K8sServiceEndpoint) EnvVars() []v1.EnvVar {
+	if k8s.Host == "" || k8s.Port == "" {
```

Comment:
Wonder if we should be doing some sort of validation on these before we pass them through to the individual components.

Seems like we should add logic to check for scenarios like this:

- An invalid port number of host is provided
- One field is provided but not the other

WDYT? We'd probably want to add that validation to core_controller.go

Reply:
> * An invalid port number of host is provided

I though about that, but can we do anything beyond "the value provided to port is a 16bit number" ? Or "is HOST parsable IP"? What if it is a DN?

> * One field is provided but not the other

It requires both to be passed on. Maybe can can do the test in the `init()` and log that one is missing and thus not passing anything on :+1:

Reply:
> Maybe can can do the test in the init() and log that one is missing and thus not passing anything on

I don't think it belongs in the init. We have an estalished pattern for doing validation and reporting it to the user via a TigeraStatus message (see SetDegraded) - we should use that same mechansim. 

> I though about that, but can we do anything beyond "the value provided to port is a 16bit number" ? Or "is HOST parsable IP"? What if it is a DN?

Yep, this is still better than nothing. We should be able to parse URL provide to make sure it is a valid IP or domain name.

Reply:
I added the validation, but note that if the provided values are wrong, operator will not work correctly, will not be able to connect. I think it will fail sooner  than these validations happen. It will appear in its logs as "cannot connect ..." I am still not sure that hardening it here adds much value except if the operator runs from outside the cluster or uses kubeconfig.

Reply:
> I added the validation, but note that if the provided values are wrong, operator will not work correctly, will not be able to connect. I think it will fail sooner than these validations happen

oof, yes you are correct.

Reply:
i am inclined to revert the last commit with the validations and assume that if operator can use these values for its own connection, then it is all OK. If not, the error will be visible with operator's pod `describe pod`

Reply:
@tomastigera that sounds good to me. Sorry for leading you astray.

Reply:
ok, no worries, we would not know if we did not do the excercise :-) I removed the validations and rebased the code on master.